controller/appcontroller.go

```diff
@@ -858,7 +863,7 @@ func (ctrl *ApplicationController) autoSync(app *appv1.Application, syncStatus *
 	// auto-sync with pruning disabled). We need to ensure that we do not keep Syncing an
 	// application in an infinite loop. To detect this, we only attempt the Sync if the revision
 	// and parameter overrides are different from our most recent sync operation.
-	if alreadyAttemptedSync(app, desiredCommitSHA) {
+	if !ctrl.setSelfHealingSequence(app, &app.Status) && alreadyAttemptedSync(app, desiredCommitSHA) {
```

Comment:
Sync might include sync hooks. The sync hooks might run e.g. database upgrade and it looks like healing attempt should not run hooks.

Instead we need to identify which resources are out-of-sync and run selective sync which exclude hooks.

Reply:
Having DB upgrades without sanity checks in a sync-hook doesn't sound like a good idea ;-)
Sync-Hooks might also trigger some monitoring systems, so you can detect or log argocd operations in your cluster. 
IMHO there are pros and cons for having self-healing and sync-hooks working together.
Removing the option to have sync-hooks and self-healing working together is in my opinion worse, than giving a heads up about possible pitfalls of this functionality in the documentation.